feature_engine/timeseries/forecasting/window_features.py

```diff
@@ -0,0 +1,333 @@
+            pandas `shift()`. It is the same functionality. If freq is a list, lag features
+            will be created for each one of the frequency values in the list. If freq is not
+            None, then this parameter overrides the parameter `periods`.
+
+        sort_index: bool, default=True
+            Whether to order the index of the dataframe before creating the lag features.
+
+        {missing_values}
+
+        {drop_original}
+
+        Attributes
+        ----------
+        variables_:
+            The group of variables that will be lagged.
+
+        {feature_names_in_}
+
+        {n_features_in_}
+
+        Methods
+        -------
+        {fit}
+
+        transform:
+            Add lag features.
+
+        {fit_transform}
+
+        See Also
+        --------
+        pandas.shift
+        """
+
+    def __init__(
+            self,
+            variables: List[str] = None,
+            window: Union[str, int] = 1,
+            function: Callable = np.mean,
+            periods: int = 1,
+            freq: str = None,
+            sort_index: bool = True,
+            missing_values: str = "raise",
+            drop_original: bool = False,
+    ) -> None:
+
+        if not (isinstance(window, int) or isinstance(window, str)) or isinstance(window, bool):
+            raise ValueError(
+                f"window must be a string or integer. Got {window} instead."
+            )
+
+        if isinstance(periods, int) and periods > 0 and not isinstance(periods, bool):
+            self.periods = periods
+        else:
+            raise ValueError(
+                f"periods must be a positive integer. Got {periods} instead."
+            )
+
+        if function not in (np.mean, np.std, np.median, np.min, np.max, np.sum):
+            raise ValueError(
+                f"function must be np.mean, np.std, np.median, np.min, np.max, or np.sum. "
+                f"Got {function} instead."
+            )
+        if periods is None and not isinstance(freq, str):
+            raise ValueError(
+                f"freq must be a string. Got {freq} instead."
+            )
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError(
+                "missing_values takes only values 'raise' or 'ignore'. "
+                f"Got {missing_values} instead."
+            )
+
+        if not isinstance(drop_original, bool):
+            raise ValueError(
+                "drop_original takes only boolean values True and False. "
+                f"Got {drop_original} instead."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.window = window
+        self.function = function
+        self.freq = freq
+        self.sort_index = sort_index
+        self.missing_values = missing_values
+        self.drop_original = drop_original
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
```

Comment:
the fit method is identical to lag features.

Could we move this to a base forecasting class? for this transformer and the lag features?